base/src/ipld.rs

```diff
@@ -21,7 +21,9 @@ pub enum Ipld {
     List(Vec<Ipld>),
     /// Represents a map.
     Map(BTreeMap<String, Ipld>),
-    /// Represents a link to an Ipld node
+    /// Represents a weak link to an Ipld node.
+    WeakLink(Cid),
```

Comment:
Based on the proposed semantics for pinning, we need to differentiate between strong and weak links. For backwards compatibility existing links will be weak, and a new encoding added for strong links.

On an ipfs website, the blocks of a page would be a strong link, while a link to a different page might be a weak one. It's up to the application developer to decide.

Reply:
A package manager would want to deal in units of packages that may be composed of multiple blocks.

Reply:
Could you please explain a bit more what a weak link is?

Reply:
The application determines what it's atoms are. So for example a package manager represents the files of a package as an atomic unit, but the metadata is also ipld and contains a home-page field. That would be a weak link that crosses the application boundaries. Although you're right, that wouldn't be a cid but a name that resolves to a cid. I'm not so sure it does make sense anymore.

Reply:
Yep, you convinced me, I don't think this makes sense after all